substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2008,6 +2008,8 @@ pub mod pallet {
 			pool_id: PoolId,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
+			// ensure pool is not in an un-migrated state.
+			ensure!(!Self::api_pool_needs_delegate_migration(pool_id), Error::<T>::NotMigrated);
```

Comment:
You can move this check to `BondedPool::<T>::get()/set()` perhaps? it doesn't work?

Reply:
This is a good idea but I am somewhat conflicted about it. Conventionally, you don't expect `get` to fail for any other reason except if key not found. 

We can change signature to return a Result<> instead of Option<>. But given that this code can be thrown away at some point, I lean towards keeping it like this.